clouds/aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,793 @@
+        try:
+            for sample in TimeoutSampler(
+                wait_timeout=300,
+                sleep=5,
+                func=is_resource_available,
+                aws_resource={"Resource": (rt_details["RouteTables"][0])["Routes"][0]},
+            ):
+                if sample is True:
+                    break
+        except TimeoutExpiredError:
+            LOGGER.error(f" Route table {rt_private_id} not available.")
+            raise
+
+        ec2_assign_name_tag(
+            ec2_client=ec2_client,
+            res_id=rt_private["RouteTable"]["RouteTableId"],
+            tag_name=f"{site_name}-rtb-private{count+1}",
+        )
+        rt_private_ids_list.append(rt_private["RouteTable"]["RouteTableId"])
+    return rt_private_ids_list
+
+
+def associate_route_ipv4(ec2_client, route_table_id, internet_gateway_id) -> None:
+    """
+    Send all IPv4 traffic to the internet gateway
+    """
+    LOGGER.info("Creating Routes for  Route Tables")
+    ec2_client.create_route(
+        RouteTableId=route_table_id,
+        DestinationCidrBlock="0.0.0.0/0",
+        GatewayId=internet_gateway_id,
+    )
+
+
+def associate_route_ipv6(ec2_client, route_table_id, internet_gateway_id) -> None:
+    """
+    Send all IPv6 traffic to the internet gateway
+
+    Args:
+         ec2_client (botocore.client.EC2): aws boto client
+         route_table_id (str): route table id string
+         internet_gateway_id (str): internet gateway id to associate
+    """
+    LOGGER.info("Creating IPV6 Routes for  Route Tables")
+    ec2_client.create_route(
+        RouteTableId=route_table_id,
+        DestinationIpv6CidrBlock="::/0",
+        GatewayId=internet_gateway_id,
+    )
+
+
+def public_route_table_association(
+    ec2_client, subnet_count, route_table_id, subnet_id
+) -> None:
+    """
+    Add public routes to the route table
+
+    Args:
+         ec2_client (botocore.client.EC2): aws boto client
+         subnet_count (int): The number of public subnets to attach
+         route_table_id: route table id string
+         subnet_id (list): list containing public subnet id's
+    """
+    LOGGER.info("Associtate public route tables")
+    for count in range(subnet_count):
+        ec2_client.associate_route_table(
+            RouteTableId=route_table_id,
+            SubnetId=subnet_id[count],
+        )
+
+
+def private_route_table_association(
+    ec2_client, subnet_count, route_table_id, subnet_id
+) -> None:
+    """
+    Add private routes to the route table
+
+    Args:
+         ec2_client (botocore.client.EC2): aws boto client
+         subnet_count (int): The number of private subnets to attach
+         route_table_id: private route table id string
+         subnet_id (list): list containing private subnet id's
+    """
+    LOGGER.info("Associtate private route tables")
+    for count in range(subnet_count):
+        ec2_client.associate_route_table(
+            RouteTableId=route_table_id[count],
```

Comment:
`route_table_id[count]` sent here but `route_table_id,` sent in `public_route_table_association` for the same function call

Reply:
for public route table there's only one id, but for private it is a list, updated docstring also to reflect it